pennylane/_qubit_device.py

```diff
@@ -1888,4 +1888,22 @@ def adjoint_jacobian(self, tape, starting_state=None, use_device_state=False):
             for kk in range(n_obs):
                 bras[kk, ...] = self._apply_operation(bras[kk, ...], adj_op)
 
+        if qml.active_return():
+            # postprocess the jacobian for the new return_type system
+            return self._adjoint_jacobian_processing(jac)
+
         return jac
+
+    @staticmethod
+    def _adjoint_jacobian_processing(jac):
+        """
+        Post-process the Jacobian matrix returned by ``adjoint_jacobian`` for
+        the new return type system.
+        """
+        if jac.shape[0] == 1:
+            if jac.shape[1] == 1:
+                return jac[0][0]
+
+            return tuple(jac[0])
+
+        return tuple(tuple(j) for j in jac)
```

Comment:
In the case of numpy array I think you need to recall it on j because currently if is returning numpy float. I had the same issue with the finite difference case.

```
qml.enable_return()
import pennylane as qml

dev = qml.device("default.qubit", wires=2)

with qml.tape.QuantumTape() as tape:
    qml.RX(0.5, wires=0)
    qml.CNOT(wires=[0, 1])
    qml.RY(0.8, wires=1)
    qml.expval(qml.PauliZ(0))
    qml.expval(qml.PauliZ(1))
    qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))

jac = dev.adjoint_jacobian(tape)
```
`((-0.479425538604203, -3.469446951953614e-18), (-0.3340189893779267, -0.6295391960392662), (0.0, -0.7173560908995227))`

Reply:
Done 👍